kpi/tests/api/v2/test_api_assets.py

```diff
@@ -1747,6 +1749,84 @@ def test_asset_redeployment(self):
+        p = dateutil.parser.parse
+
+        assert self.asset.date_deployed is None
+        asset_response = self.client.get(self.asset_url, format='json')
+        assert asset_response.data['date_deployed'] is None
+
+        before = timezone.now()
+        self.test_asset_deployment()
+        after = timezone.now()
+
+        asset_response = self.client.get(self.asset_url, format='json')
+        date_first_deployed = asset_response.data['date_deployed']
+        assert before <= p(date_first_deployed) <= after
+        deployed_version = asset_response.data['deployed_versions']['results'][
+            0
+        ]
+        assert asset_response.data['version_id'] == deployed_version['uid']
+        assert(
+            asset_response.data['deployed_version_id']
+            == deployed_version['uid']
+        )
+
+        # Add a form media file to this asset
+        crab_png_b64 = (
+            'iVBORw0KGgoAAAANSUhEUgAAABIAAAAPAgMAAACU6HeBAAAADFBMVEU7PTqv'
+            'OD/m6OX////GxYKhAAAAR0lEQVQI1y2MMQrAMAwD9Ul5yJQ1+Y8zm0Ig9iur'
+            'kmo4xAmEUgJpaYE9y0VLBrwVO9ZzUnSODidlthgossXf73pNDltav88X3Ncm'
+            'NcRl6K8AAAAASUVORK5CYII='
+        )
+        asset_file_list_url = reverse(
+            self._get_endpoint('asset-file-list'), args=[self.asset.uid]
+        )
+        asset_file_post_data = {
+            'file_type': AssetFile.FORM_MEDIA,
+            'description': 'I have pincers',
+            'base64Encoded': 'data:image/png;base64,' + crab_png_b64,
+            'metadata': json.dumps({'filename': 'crab.png'}),
+        }
+        asset_file_response = self.client.post(
+            asset_file_list_url, asset_file_post_data
+        )
+        assert asset_file_response.status_code == status.HTTP_201_CREATED
+
+        # Redeploy with the new media file, which is a change that occurs
+        # without creating a new `AssetVersion`
+        deployment_url = reverse(
+            self._get_endpoint('asset-deployment'),
+            kwargs={'uid': self.asset_uid},
+        )
+        before = timezone.now()
+        redeploy_response = self.client.patch(
+            deployment_url,
+            {
+                'backend': 'mock',
+                'active': True,
+                'version_id': deployed_version['uid'],
+            },
+        )
+        after = timezone.now()
+        assert redeploy_response.status_code == status.HTTP_200_OK
+
+        asset_response = self.client.get(self.asset_url, format='json')
+
+        assert (
+            before <= p(asset_response.data['date_deployed']) <= after
+        ), 'Redeployment should update the deployment timestamp'
+
+        assert (
+            deployed_version['date_modified']
+            == asset_response.data['deployed_versions']['results'][0][
+                'date_modified'
+            ]
+        ), (
+            'Redeploying the same version, unmodified, should not change the'
+            ' modification date of that version'
+        )
+
```

Comment:
If you agree with my comments above, I would also add a test to validate the `date_modified` stays the same on archiving.

Reply:
Done!